lib/webtask.js

```diff
@@ -708,6 +747,49 @@ exports.ensure_authenticated = function(ctx) {
             });
         }
 
+        function validate_auth_token(auth) {
+            var auth_token;
+            var decoded_auth_token;
+
+            try {
+                auth_token = req.query.key || (req.headers['authorization']
+                    .match(/\s.+$/)[0]
+                    .replace(/\s/, ''));
+            } catch(e) {
+                throw new Error('Could not parse authorization header.');
+            }
+
+            if(auth.secret) {
+               return jsonwebtoken.verify(auth_token, auth.secret);
+            } else {
+                var secret = new Buffer(AUTH0_CLIENT_SECRET, 'base64');
+
+                var userinfo = jsonwebtoken.verify(auth_token, secret);
+
+                if(!auth.email) {
+                    return userinfo;
+                } else {
+                    if(userinfo.email_verified === false) {
+                        throw new Error('User email inverified');
+                    }
+
+                    if(auth.email instanceof Array) {
+                        for(var i = 0; i < auth.email.length; i++)
+                            if(userinfo.email.match(auth.email[i])) {
+                                return userinfo;
+                            }
+                    }
+                    else {
+                        if(userinfo.email.match(auth.email)) {
+                            return userinfo;
+                        }
+                    }
+
+                    throw new Error('User email unauthorized.');
```

Comment:
Same here. "The account you used to login (`userinfo.email`) does not match the authorization policies"